### DISCUSSION TITLE
Import / path to repo keyfile

### DISCUSSION THREAD

**Discussion by dtu-compute** [borgbase/vorta#803] [Ideas]
**Is your feature request related to a problem? Please describe.**
A way to import an existing repo when a keyfile is not stored in the repo, but on the workstation.

**Describe the solution you'd like**
1. the UI should have a text field with says where the keyfile is located on the workstation. I have no idea where it is stored.
2. the UI should allow for change the keyfile. E.g. in the case where I need to add an existing repo. Or in the recovery case, where the workstation have been completely reinstalled and Vorta have to be configured to restore the remote files.

**Comment by Hofer-Julian** [comment]
Supporting external keyfiles might be a good idea. If we would, the question would be how far the support should go.

Only support existing repos with external keyfiles?
Support to create new ones with external keyfiles?
Allow to modify the keyfile?

What do you say @m3nu?

**Comment by lightbulbjim** [comment]
All three? :-D

I'd love to see this supported. At the moment if the repository link is lost (either by fat fingers or in a DR scenario) then the only option is to recreate it from scratch.

**Comment by m3nu** [comment]
Let's file this as idea. In general keyfiles are more dangerous than repokeys and we don't recommend them anywhere.

**Comment by samuel-w** [comment]
Currently I have a semiworking key file import/export system set up. https://github.com/samuel-w/vorta-1/tree/KeyManagement
Should I add the ability to import keys made with `borg key import --paper`? The only way I see to implement this is to copy the entire paper key validation method from borg, as the input for it is interactive. Another way could be to implement a mini console in Qt that runs the command, but it wouldn't match the rest of the UI and be a bit confusing. 

I currently have it set up to combine key import/export and password changes in the same place, as they all have the same root `borg key` command.
![keymanagement](https://user-images.githubusercontent.com/67625312/87891148-97919c80-c9fe-11ea-9255-b5b634127c18.png)
